src/components/Bullets.js

```diff
@@ -0,0 +1,74 @@
+import * as React from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+import { PALETTE } from "../shared";
+
+const ListEl = styled.li`
+  position: relative;
+  margin: 0.5rem 0;
+
+  &::before {
+    content: "";
+    position: absolute;
+    left: -1.4rem;
+    top: 0.7rem;
+    width: 6px;
+    height: 6px;
+    border-radius: 50%;
+    background-color: ${PALETTE.black60};
+  }
+
+  &:first-child > ul > & {
+    &::before {
+      width: 4px;
+      height: 4px;
+      border: 1px solid ${PALETTE.black60};
+      background-color: transparent;
+    }
+  }
+
+  &:nth-child(2) > ul > & {
+    &::before {
+      width: 6px;
+      height: 6px;
+      border-radius: initial;
+      background-color: ${PALETTE.black60};
+    }
+  }
+`;
+
+const El = styled.ul`
+  position: relative;
+  list-style-type: none;
+  color: ${PALETTE.black60};
+  padding-left: 1.375rem;
+`;
+
+/**
+ * Note: This exports a React component instead of a styled-component.
+ * [Design Mockup](https://zpl.io/2ZXQ0YJ)
+ */
+
+const Bullets = ({ data }) => (
+  <El>
+    {data.map((datum, index) => {
+      if (datum.subBullets) {
+        return (
+          <ListEl>
```

Comment:
You have to be careful about components that accept a very specific data shape, because it can be difficult for people to follow. If we do end up doing this, we should consider porting this library to Typescript so that people can see the documentation in their editor.

An alterative is to let people type in `<li>`s as the child elements:

```
<Bullets>
  <li>The first line of the list</li>
  <li>We can even support other React components this way: <ListComponent /></li>
  <li>
    You can nest things like so
    <Bullets>
      <li>I am nested</li>
    </Bullets>
  </li>
</Bullets>
```

When your component designs resonate with the way people do things ordinarily, it's easier and faster for people to understand them, because they have past experience to hang onto. Something to consider!

Reply:
my recursion :(